### ISSUE TITLE
Bad caret placement when using `lZero` and entering a `0` on a value superior to 1.

### ISSUE THREAD

**Issue by AlexandreBonneau** [BobKnothe/autoNumeric#283]
When using `lZero: 'deny'` and entering a `0` on the far left of the input value which has a value superior to 1, the caret is moved on the far right.

For instance, using those options : `{ aSep: ' ', aDec: ',', altDec: '.', aSign: '%', pSign: 's', mRound: 'U', mDec: '3', lZero: 'deny' }`, if I input `1,234` into the input, then move the caret of the far left and input `0`, then the caret is moved of the far right of the value, while it should stay in place.

**Comment by AlexandreBonneau** [comment]
Additionally, when using the following options :
```
let euroPositive = { aSep: '.', aDec: ',', altDec: '.', aSign: '\u00a0€', pSign: 's', mRound: 'U', vMin: '0', lZero: 'deny' }
```

Then if you have `8.000 €` written in the output, and try to input `0` when the cursor is on the far left, then your cursor jumps to `8.0|00 €`.

**Comment by AlexandreBonneau** [comment]
Here is the related [codepen](http://codepen.io/AnotherLinuxUser/pen/Wodaoo) which shows this weird behavior.

**Comment by BobKnothe** [comment]
Here is a fix for this issue

Need to create tests confirming

        setValueParts(left, right, advent) {
            const settingsClone = this.settingsClone;
            const parts = this.normalizeParts(left, right);
            const [minTest, maxTest] = autoCheck(this.newValue, settingsClone);
            let position = parts[0].length;
            this.newValue = parts.join('');

            if (minTest && maxTest) {
                this.newValue = truncateDecimal(this.newValue, settingsClone, advent);
                const testValue = (contains(this.newValue, ',')) ? this.newValue.replace(',', '.') : this.newValue;
                if (testValue === '' || testValue === settingsClone.aNeg) {
                    settingsClone.rawValue = '';
                } else {
                    settingsClone.rawValue = testValue;
                }
                if (position > this.newValue.length) {
                    position = this.newValue.length;
                }
				
				// fix for issue #283 
                if (position === 1 && parts[0] === '0' && settingsClone.lZero === 'deny') {
                    position = 0;
                }
                this.value = this.newValue;
                this.setPosition(position, false);
                return true;
            }

            if (!minTest) {
                this.$that.trigger('autoNumeric:minExceeded');
            } else if (!maxTest) {
                this.$that.trigger('autoNumeric:maxExceeded');
            }

            return false;
        },
